functions/Invoke-DbaDiagnosticQuery.ps1

```diff
@@ -0,0 +1,278 @@
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.	
+
+.LINK
+https://dbatools.io/Invoke-DbaDiagnosticQuery
+
+
+.EXAMPLE   
+Invoke-DbaDiagnosticQuery -SqlInstance sql2016 -UseSelectionHelper
+
+Provides a gridview with all the queries to choose from and will run the selection made by the user on the Sql Server instance specified. 
+
+#>
+	
+	[CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess = $true)]
+	param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlServer")]
+		[DbaInstanceParameter[]]$SqlInstance,
+		[System.IO.FileInfo]$Path,
+		[string[]]$QueryName,
+		[switch]$UseSelectionHelper,
+		[switch]$InstanceOnly,
+		[switch]$DatabaseSpecific,
+		[switch]$Silent
+	)
+	
+	begin {
+		
+		function Invoke-DiagnosticQuerySelectionHelper {
+			Param (
+				[parameter(Mandatory = $true)]
+				$ParsedScript
+			)
+			
+			[string[]]$selection = $script | Select-Object QueryNr, QueryName, DBSpecific, Description | Out-GridView -Title "Diagnostic Query Overview" -OutputMode Multiple | Sort-Object QueryNr | Select-Object -ExpandProperty QueryName
+			
+			$selection
+		}
+		
+		Write-Message -Level Verbose -Message "Interpreting DMV Script Collections"
+		
+		$module = Get-Module -Name dbatools
+		$base = $module.ModuleBase
+		
+		if (!$path) {
+			$path = "$base\bin\diagnosticquery"
+		}
+		
+		$scriptversions = @()
+		$scriptfiles = Get-ChildItem "$path\SQLServerDiagnosticQueries_*_*.sql"
+		
+		if (!$scriptfiles) {
+			Write-Message -Level Warning -Message "Diagnostic scripts not found in $path. Using the ones within the module."
+			
+			$path = "$base\bin\diagnosticquery"
+			
+			$scriptfiles = Get-ChildItem "$base\bin\diagnosticquery\SQLServerDiagnosticQueries_*_*.sql"
+			if (!$scriptfiles) {
+				Stop-Function -Message "Unable to download scripts, do you have an internet connection? $_" -InnerErrorRecord $_
+			}
+		}
+		
+		[int[]]$filesort = $null
+		
+		foreach ($file in $scriptfiles) {
+			$filesort += $file.BaseName.Split("_")[2]
+		}
+		
+		$currentdate = $filesort | Sort-Object -Descending | Select-Object -First 1
+		
+		foreach ($file in $scriptfiles) {
+			if ($file.BaseName.Split("_")[2] -eq $currentdate) {
+				$script = Invoke-DbaDiagnosticQueryScriptParser -filename $file.fullname
+				
+				$newscript = [pscustomobject]@{
+					Version = $file.Basename.Split("_")[1]
+					Script = $script
+				}
+				$scriptversions += $newscript
+			}
+		}
+	}
+	
+	process {
+		
```

Comment:
Add an `if (Test-FunctionInterrupt) { return }` to react to failure in the begin block